addons/storagecluster_controller.go

```diff
@@ -0,0 +1,139 @@
+package addons
+
+import (
+	"context"
+	"log/slog"
+	"reflect"
+
+	multiclusterv1alpha1 "github.com/red-hat-storage/odf-multicluster-orchestrator/api/v1alpha1"
+	"github.com/red-hat-storage/odf-multicluster-orchestrator/controllers/utils"
+
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v4/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// StorageClusterReconciler reconciles a MirrorPeer object
+type StorageClusterReconciler struct {
+	Scheme           *runtime.Scheme
+	HubClient        client.Client
+	SpokeClient      client.Client
+	SpokeClusterName string
+	Logger           *slog.Logger
+
+	CurrentNamespace string
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *StorageClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	scToMirrorPeerMapFunc := func(ctx context.Context, obj client.Object) []ctrl.Request {
+		reqs := []ctrl.Request{}
+		var mirrorPeerList multiclusterv1alpha1.MirrorPeerList
+		err := r.HubClient.List(ctx, &mirrorPeerList)
+		if err != nil {
+			r.Logger.Error("Unable to reconcile MirrorPeer based on storagecluster annotation changes.", "error", err)
+			return reqs
+		}
+		for _, mirrorpeer := range mirrorPeerList.Items {
+			if mirrorpeer.Status.Phase == multiclusterv1alpha1.IncompatibleVersion {
+				continue
+			}
+			reqs = append(reqs, reconcile.Request{NamespacedName: types.NamespacedName{Name: mirrorpeer.Name}})
+		}
+		return reqs
+	}
+
+	scPredicate := predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			return false
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return false
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			oldSC, ok := e.ObjectOld.(*ocsv1.StorageCluster)
+			if !ok {
+				return false
+			}
+			newSC, ok := e.ObjectNew.(*ocsv1.StorageCluster)
+			if !ok {
+				return false
+			}
+			if !reflect.DeepEqual(oldSC.Annotations, newSC.Annotations) {
+				return true
+			}
+
+			return false
+		},
+		GenericFunc: func(_ event.GenericEvent) bool {
+			return false
+		},
+	}
```

Comment:
I think we have a AnnotationChangedPredicate already defined in the predicate pkg https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/predicate#AnnotationChangedPredicate